go/runtime/registry/registry.go

```diff
@@ -388,11 +388,26 @@ func (r *runtime) run(ctx context.Context) {
 		select {
 		case <-ctx.Done():
 			return
-		case <-epoCh:
+		case epoch := <-epoCh:
 			if up := r.updateActiveDescriptor(ctx); up && !activeInitialized {
 				close(r.activeDescriptorCh)
 				activeInitialized = true
 			}
+
+			// Trigger clean-up for bundles less than active version.
```

Comment:
We will need to move this code and the code below that triggers downloads to the manager once that is possible (i.e. when there are no more dependencies). Maybe this is already possible :thinking: However, in the future the manager should register to active and registry descriptor events (maybe only to the latter), and trigger cleanup and download when needed.

Reply:
Agree, will move in another PR, just mentioning here.